Added IAnimatedVisual2 interface and Freeze/Unfreeze AVP methods

aborziak-ms opened the pull request:
## Description
This PR adds new interface `IAnimatedVisual2`, this interface provides methods that allow to destroy or re-instantiate all the animations that are used by animated visual. This can help reduce the CPU overhead when animated visual is idle.

Also exposing new methods Freeze/Unfreeze/CanFreeze on `AnimatedVisualPlayer` that allows to control behavior of the animated visual.

### This is currently a draft

StephenLPeters commented on dev/AnimatedVisualPlayer/AnimatedVisualPlayer.idl:
```diff
@@ -62,6 +80,9 @@ unsealed runtimeclass AnimatedVisualPlayer
     Boolean AutoPlay;
+    [MUX_DEFAULT_VALUE("true")]
+    [MUX_PROPERTY_CHANGED_CALLBACK(TRUE)]
+    Boolean Active;
```
This will need to be marked as Preview as well.

aborziak-ms replied:
Not sure what the syntax should be, I tried just `[MUX_PREVIEW]` or `[MUX_PREVIEW] { ... }` but it resulted in runtime error:
![image](https://user-images.githubusercontent.com/83784664/139968448-bd7eef76-5ae8-4f00-a84a-4e87bee07a73.png)

StephenLPeters commented on dev/AnimatedVisualPlayer/AnimatedVisualPlayer.cpp:
```diff
@@ -932,6 +981,11 @@ void AnimatedVisualPlayer::UpdateContent()
+
+    if (!Active())
+    {
+        Pause();
```
Is this the correct place to check for Active state? right above we've called start, should it be in the check there?  Why do we even need to do this, if the AVP isn't active we don't ahve any animations so why do we need to pause?

aborziak-ms replied:
If AutoPlay is true we will start an animation on root.Progress property (a few lines above). Here I'm pausing this animation since there is nothing to animate yet. It will unpause when we set Active = true

aborziak-ms replied:
Changed it so that if Active==false, we will set AutoPlay to false (and will destroy any m_nowPlaying if it exists)

aborziak-ms commented:
@StephenLPeters I think with my last changes I was able to fix flickering issue. So the idea is that when we are setting `Active` to `false`, instead of `Pause()` we will call `Stop()`, and then (important) will wait for `compositor.Commit` callback. It should ensure that we updated the progress in compositor before calling `DestroyAnimations()`.

Then I also noticed that there was a variable `m_currentPlayFromProgress` that was never changed in old code, I renamed it and added more uses, now this variable should track last known progress value. When we are starting new play, `m_lastPlayProgress` will be set to `fromProgress`, and when it completes it will be set to `toProgress`. It will allow us to use `m_lastPlayprogress` as a hint for Lottie, because if we know last progress value when we called `Active=false`, we can re-instantiate animations without flickering.

And for that we also need one new parameter: `InstantiateAnimations(Double progressHint)`.

P.S. I also updated test page and LottieLogo animation.
